profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSamplerLoop.cpp

```diff
@@ -587,22 +614,57 @@ void StackSamplerLoop::PersistStackSnapshotResults(StackSnapshotResultBuffer con
 
     if (_pConfiguration->IsFFLibddprofEnabled())
     {
-        // add the snapshot to the lipddprof pipeline
+        // add the WallTime sample to the lipddprof pipeline
         WallTimeSampleRaw rawSample;
         rawSample.Timestamp = pSnapshotResult->GetUnixTimeUtc();
-        rawSample.Duration = pSnapshotResult->GetRepresentedDurationNanoseconds();
         rawSample.LocalRootSpanId = pSnapshotResult->GetLocalRootSpanId();
         rawSample.SpanId = pSnapshotResult->GetSpanId();
         rawSample.AppDomainId = pSnapshotResult->GetAppDomainId();
         pSnapshotResult->CopyInstructionPointers(rawSample.Stack);
         rawSample.ThreadInfo = pThreadInfo;
         pThreadInfo->AddRef();
-
+        rawSample.Duration = pSnapshotResult->GetRepresentedDurationNanoseconds();
         _pWallTimeCollector->Add(std::move(rawSample));
+
+
+        // add the CPU sample to the lipddprof pipeline if needed
```

Comment:
The sample shouldn't be based on whether it _did_ run on CPU, as it may now be stuck in a syscall or unscheduled. The sampling should however be based on whether it is _currently_ running on CPU. On Linux, you check that based on whether the state of the thread is `RUNNING`. Is there an equivalent on Windows?

Reply:
You can't do that even on Linux because you just suspended it to walk its stack  :^)

Reply:
And on Linux, you check the `/proc/self/task/{tid}/stat` to know whether the thread is running.

Reply:
I will have a look to see how to check this on Windows + Linux implementation